src/proxy-animation.js

```diff
@@ -912,6 +906,147 @@ function autoAlignStartTime(details) {
   }
 }
 
+function unsupportedTimeline(timeline) {
+  throw new Error('Unsupported timeline class');
+}
+
+function getNormalStartRange(timeline) {
+  if (timeline instanceof ViewTimeline) {
+    return { rangeName: 'cover', offset: CSS.percent(0) };
+  }
+
+  if (timeline instanceof ScrollTimeline) {
+    return CSS.percent(0);
+  }
+
+  unsupportedTimeline(timeline);
+}
+
+function getNormalEndRange(timeline) {
+  if (timeline instanceof ViewTimeline) {
+    return { rangeName: 'cover', offset: CSS.percent(100) };
+  }
+  
+  if (timeline instanceof ScrollTimeline) {
+    return CSS.percent(100);
+  }
+
+  unsupportedTimeline(timeline);
+}
+
+function parseAnimationRange(timeline, value) {
+  const animationRange = {
+    start: getNormalStartRange(timeline),
+    end: getNormalEndRange(timeline),
+  };
+
+  if (!value)
+    return animationRange;
+
+  if (timeline instanceof ViewTimeline) {
+    // Format:
+    // <start-name> <start-offset> <end-name> <end-offset>
+    // <name> --> <name> 0% <name> 100%
+    // <name> <start-offset> <end-offset> --> <name> <start-offset>
+    //                                        <name> <end-offset>
+    // <start-offset> <end-offset> --> cover <start-offset> cover <end-offset>
+    // TODO: Support all formatting options once ratified in the spec.
+    const parts = value.split(' ');
+    const rangeNames = [];
+    const offsets = [];
+  
+    parts.forEach(part => {
+      if (part.endsWith('%'))
```

Comment:
To support `calc()`, maybe this could be replaced by 

```js
if (ANIMATION_RANGE_NAMES.includes(part)) 
  rangeNames.push(part);
} else {
  offsets.push(CSSNumericValue.parse(part));
}
```

Then the following code can be removed: 

```js
    // TODO: allow calc() in the offsets
     if (offsets.length > 1) {
       animationRange.start.offset = CSS.percent(offsets[0]);
       animationRange.end.offset = CSS.percent(offsets[1]);
     }
```